blocks/graph.py

```diff
@@ -37,6 +37,7 @@ def _get_variables(self):
         self.applies = set()
         self.application_calls = set()
         self.updates = []
+        self.auxiliary_variables = set()
```

Comment:
Well, no reasonable brick will declare L2 penalty as an auxiliary variable, since it is too generic and to easy to evaluate in a way your describe above. The auxiliary variables is a mechanism for more involved cases, when there exists a highly specialized local variable that is hard to compute otherwise, like Jan's penalty.